pkg/model/storage.go

```diff
@@ -1,49 +1,95 @@
 
 type S3Storage struct {
-	// The root path for the backup repository (just a path, without bucket)
+	// Path is the root directory within the S3 bucket where backups will be stored.
+	// It should not include the bucket name.
 	Path string
-	// The S3 bucket
+	// Bucket is the name of the S3 bucket where backups will be stored.
 	Bucket string
-	// The S3 region string (AWS S3 optional).
+	// S3Region is the AWS region where the S3 bucket is located.
 	S3Region string
-	// The S3 profile name (AWS S3 optional).
+	// S3Profile is the name of the AWS credentials profile to use.
 	S3Profile string
-	// An alternative endpoint for the S3 SDK to communicate (AWS S3 optional).
+	// S3EndpointOverride is used to specify a custom S3 endpoint.
 	S3EndpointOverride *string
-	// The log level of the AWS S3 SDK (AWS S3 optional).
+	// S3LogLevel controls the verbosity of the AWS SDK logging.
+	// Valid values are: OFF, FATAL, ERROR, WARN, INFO, DEBUG, TRACE.
 	S3LogLevel *string
-	// The minimum size in bytes of individual S3 UploadParts
+	// MinPartSize is the minimum size in bytes for multipart upload parts.
 	MinPartSize int
-	// The maximum number of simultaneous requests from S3.
+	// MaxConnsPerHost limits the number of concurrent connections to S3.
 	MaxConnsPerHost int
 }
 
 func (s *S3Storage) storage() {}
 
 type GcpStorage struct {
-	// Path to file containing Service Account JSON Key.
+	// KeyFile is the path to the JSON file containing the Google Cloud service account key.
+	// This file is used for authentication with GCP services.
 	KeyFile string
-	// For GCP storage bucket is not part of the path as in S3.
-	// So we should set it separately.
+	// BucketName is the name of the GCP bucket where backups will be stored.
 	BucketName string
-	// The root path for the backup repository.
+	// Path is the root directory within the GCS bucket where backups will be stored.
 	Path string
-	// Alternative url.
-	// It is not recommended to use an alternate URL in a production environment.
+	// Endpoint is an alternative URL for the GCS API.
+	// This should only be used for testing or in specific non-production scenarios.
 	Endpoint string
 }
 
 func (s *GcpStorage) storage() {}
+
+// AzureStorage represents the configuration for Azure Blob storage.
+type AzureStorage struct {
+	// Path is the root directory within the Azure Blob container where backups will be stored.
+	Path string
+	// Endpoint is the URL of the Azure Blob storage service.
+	Endpoint string
+	// ContainerName is the name of the Azure Blob container where backups will be stored.
+	ContainerName string
+	// Auth holds the authentication details for Azure Blob storage.
+	// It can be either AzureSharedKeyAuth or AzureADAuth.
+	Auth AzureAuth
+}
+
+func (s *AzureStorage) storage() {}
+
+// AzureAuth represents the authentication methods for Azure Blob storage.
+// This interface is implemented by AzureSharedKeyAuth and AzureADAuth.
+type AzureAuth interface {
+	azureAuth()
+}
```

Comment:
Consider using the constraint interface:
```go
type AzureAuth interface {
	AzureSharedKeyAuth | AzureADAuth
}
```
but this will require additional changes in the current logic.

Reply:
```go
type AzureStorage[T AzureAuth] struct {
	// Path is the root directory within the Azure Blob container where backups will be stored.
	Path string
	// Endpoint is the URL of the Azure Blob storage service.
	Endpoint string
	// ContainerName is the name of the Azure Blob container where backups will be stored.
	ContainerName string
	// Auth holds the authentication details for Azure Blob storage.
	// It can be either AzureSharedKeyAuth or AzureADAuth.
	Auth T
}
```
But more changes will be required.

Reply:
Can be revisited later.